Ghidra/Processors/68000/data/languages/68000.sinc

```diff
@@ -700,6 +700,100 @@ macro extendedResultFlags(result) {
   ZF = (result == 0) && (ZF == 1);
 }
 
+macro arithmeticShiftLeft(count, register, width) {
+	local modcount = count % 64;
+	CF = 0;
+	resflags(register);
+	if (count == 0) goto inst_next;
+	local msbBefore:4 = zext(register) >> (width - 1);
+	getbit(CF, register, width - modcount);
+	register = register << modcount;
+	resflags(register);
+	local msbAfter:4 = zext(register) >> (width - 1);
+	VF = (msbBefore ^ msbAfter) != 0;
+	XF = CF;
+}
+
+macro arithmeticShiftRight(count, register, width) {
+	local modcount = count % 64;
+	CF = 0;
+	resflags(register);
+	if (count == 0) goto inst_next;
+	local msbBefore:4 = zext(register) >> (width - 1);
+	getbit(CF, register, modcount - 1);
+	register = register s>> modcount;
+	resflags(register);
+	local msbAfter:4 = zext(register) >> (width - 1);
+	VF = (msbBefore ^ msbAfter) != 0;
+	XF = CF;
+}
+
+macro logicalShiftLeft(count, register, width) {
+	local modcount = count % 64;
+	CF = 0;
+	VF = 0;
+	resflags(register);
+	if (modcount == 0) goto inst_next;
+	getbit(CF, register, width - modcount);
+	register = register << modcount;
+	resflags(register);
+	XF = CF;
+}
+
+macro logicalShiftRight(count, register, width) {
+	local modcount = count % 64;
+	CF = 0;
+	VF = 0;
+	resflags(register);
+	if (modcount == 0) goto inst_next;
+	getbit(CF, register, width - modcount);
+	register = register >> modcount;
+	resflags(register);
+	XF = CF;
+}
+
+macro rotateLeft(count, register, width) {
+	local modcount = (count % 64) % width;
+	VF = 0;
+	register = (register << modcount) | (register >> (width - modcount));
+	getbit(CF, register, width - 1);
+	resflags(register);
+}
+
+macro rotateRight(count, register, width) {
+	local modcount = (count % 64) % width;
+	VF = 0;
+	register = (register << (width - modcount)) | (register >> modcount);
+	getbit(CF, register, 0);
+	resflags(register);
+}
+
+macro rotateLeftExtended(count, register, width) {
+	local modcount = (count % 64) % width;
+	CF = XF;
+	VF = 0;
+	resflags(register);
+	if (modcount == 0) goto inst_next;
+	local xflag = (register & (1 << (width - modcount))) != 0;
+	register = (register << modcount) | (zext(XF) << (modcount - 1)) | (register >> (width - modcount + 1));
+	XF = xflag;
+	CF = XF;
+	resflags(register);
+}
+
+macro rotateRightExtended(count, register, width) {
+	local modcount = (count % 64) % width;
+	CF = XF;
+	VF = 0;
+	resflags(register);
+	if (modcount == 0) goto inst_next;
+	local xflag = (register & (1 << (modcount - 1))) != 0;
```

Comment:
Unlike rotateLeft, the `xflag` calc would be:
`local xflag = ((register >> 1 | XF << width ) & (1 >> (xmod - 1)) != 0;`
again, if register is 01234567 and XF is 8. If your width is 8 and your modcount is 9, that would be:
`01234567 | (8 << 8) = 801234567`
`modcount = 9`: ` 801234567 & (1 << (9 - 1) = 80000000`
`modcount = 8`: ` 801234567 & (1 << (8 - 1) = 00000000`
`modcount = 10`: ` 801234567 & (1 << (1 - 1) = 00000007`
`modcount = 11`: ` 801234567 & (1 << (2 - 1) = 00000060`
... etc